Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp

```diff
@@ -1290,14 +1319,15 @@ void MediaPlayerPrivateGStreamerBase::handleProtectionStructure(const GstStructu
 void MediaPlayerPrivateGStreamerBase::cdmInstanceAttached(const CDMInstance& instance)
 {
     ASSERT(isMainThread());
+    LockHolder lock(m_protectionMutex);
     GST_DEBUG("CDM instance %p set", &instance);
     m_cdmInstance = &instance;
-    dispatchLocalCDMInstance();
+    dispatchLocalCDMInstance(false);
```

Comment:
I prefer to do it like this to avoid any possible race condition between unlocking and dispatching. Difficult but possible and this makes it impossible.